org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLCompletionSnippetsTest.java

```diff
@@ -537,4 +537,39 @@ public void emptyCompletionInsideDOCTYPE() throws BadLocationException {
 		testCompletionFor("<!DOCTYPE root-element |SYSTEM \"file.dtd\">", 0);
 		testCompletionFor("<!DOCTYPE root-element SYSTEM |\"file.dtd\">", 0);
 	}
+
+	// Tests triggering completion at the very beginning of the file.
+	// Should be possible if not in front of the xml declaration.
+	@Test

```

Comment:
Add a test case for completion at the beginning for a document with just the xml declaration:
![xsi-appear-before-xml-decl](https://user-images.githubusercontent.com/22376627/83420063-0c2f5e00-a3f4-11ea-8237-3262af92b61a.gif)
I think it might be a good idea to also add a test case with a completely empty file.

Reply:
> 
> 
> Add a test case for completion at the beginning for a document with just the xml declaration:
> ![xsi-appear-before-xml-decl](https://user-images.githubusercontent.com/22376627/83420063-0c2f5e00-a3f4-11ea-8237-3262af92b61a.gif)

woops, yes.

> I think it might be a good idea to also add a test case with a completely empty file.

I think this is covered sufficiently by the empty xml test:
https://github.com/eclipse/lemminx/blob/c97206e6de3033f21445401b29d9a651dde6cd95/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLCompletionSnippetsTest.java#L37-L137

Reply:
Okay. Sorry, I didn't know that checks for empty documents were already there. I should have checked first.

Reply:
no worries!

changed the rest, though.